lib/chef/resource/windows_certificate.rb

```diff
@@ -230,13 +264,15 @@ def export_cert(cert_obj, cert_path)
           out_file.close
         end
 
-        def cert_location
-          @location ||= new_resource.user_store ? "CurrentUser" : "LocalMachine"
+        # this array structure is solving 2 problems. The first is that we need to have support for both the CurrentUser AND LocalMachine stores
+        # Secondly, we need to pass the proper constant name for each store to win32-certstore but also pass the short name to powershell scripts used here
+        def get_cert_location
+          new_resource.user_store ? ["CurrentUser", CERT_SYSTEM_STORE_CURRENT_USER] : ["LocalMachine", CERT_SYSTEM_STORE_LOCAL_MACHINE]
         end
 
         def cert_script(persist)
           cert_script = "$cert = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2"
-          file = Chef::Util::PathHelper.cleanpath(new_resource.source)
+          file = Chef::Util::PathHelper.cleanpath(new_resource.source, get_cert_location[1])
```

Comment:
You've got magic `0` and `1` numbers here which makes for fairly poor reading.  I also don't like the `get_` prefix (there's actually a rubocop rule to not do this, but we haven't enabled it because we violate it so much, but I do sort of agree that we should avoid that where possible).

So, I'd suggest something like:

```
def cert_location_name
  new_resource.user_store ? "CurrentUser" : "LocalMachine"
end

def cert_location_symbol
  new_resource.user_store ? CERT_SYSTEM_STORE_CURRENT_USER : CERT_SYSTEM_STORE_LOCAL_MACHINE
end
```

That is more verbose in the declaration, but it leads to a lot easier to read with the substitutions `cert_location_name` for `get_cert_location[0]` and `cert_location_symbol` for `get_cert_location[1]`.

Reply:
The powershell code in here needs to see CurrentUser instead of CERT_SYSTEM_STORE_CURRENT_USER

Reply:
done, thanks